kafka/README.md

```diff
@@ -65,7 +67,38 @@ channels:
         topic: 'my-specific-topic-name'
         partitions: 20
         replicas: 3
-        bindingVersion: '0.3.0'
+        topicConfiguration:
+          cleanup.policy: ["delete", "compact"]
+          retention.ms: 604800000
+          retention.bytes: 1000000000
+          delete.retention.ms: 86400000
+          max.message.bytes: 1048588
+        bindingVersion: '0.4.0'
+```
+<a name="topicConfiguration"></a>
+## TopicConfiguration Object
+
+This objects contains information about the API relevant topic configuration in Kafka.
+
+Field Name |  Type   |                                                                             Description                                                                              | Applicability [default] | Constraints
+---|:-------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------:|:-----------------------:|---
+<a name="topicConfigurationCleanupPolicy"></a>`cleanup.policy` |  array  |                            The [`cleanup.policy`](https://kafka.apache.org/documentation/#topicconfigs_cleanup.policy) configuration option.                          |        OPTIONAL         | array may only contain `delete` and/or `compact`
```

Comment:
I think the other four config properties make total sense to include, as they are properties that can cause problems for an application if they don't behave in a compatible way. 

But I'm not as sure about this one. Why do you think client applications need to know about the cleanup policy?

Reply:
The cleanup policy defines if a client can expect to receive every record written to the topic. 

`compact`: A client can expect to receive the last record for a specific key, no matter if it has been written yesterday of 2 years ago. A client has to cope with the fact that he may only get the last record for that specific key.

`delete`: A client can expect to get all records written before `retension.ms` or `retension.bytes`. The client can expect to get all records in that timeframe independent of the record key.

This information is important for the design of applications consuming topics